lua/sp.c

```diff
@@ -6642,10 +6650,17 @@ static int emit_result(Lua L, long long *sprc, char **err)
 
     if (rettab) dbtable_emit_int(L, rettab);
 
-    if (retnum && (retnum < -299 || retnum > -200)) retnum = -200;
+    if (retnum && (retnum < -299 || retnum > -200)) {
```

Comment:
This seems wrong - we're catching out-of-range return code. Error message needs to reflect that (suggestion: `attempt to return out of range values:%d`).